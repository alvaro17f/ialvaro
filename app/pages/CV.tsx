import { domAnimation, LazyMotion, m } from "framer-motion";

export default function CV() {
	return (
		<>
			<LazyMotion features={domAnimation}>
				<section id="cv">
					<div className="h-24" />
					<a href="/cv/cv.pdf" target="_blank" rel="noopener noreferrer">
						<m.div
							className="grid place-items-center p-5 mb-5 rounded-xl border-[5px] border-alvaro-danger text-alvaro-white"
							initial={{ opacity: 0 }}
							whileInView={{ opacity: 1 }}
							whileHover={{ scale: 1.02, transition: { duration: 0.3 } }}
							whileTap={{ scale: 0.99, transition: { duration: 0.1 } }}
							viewport={{ once: false }}
							transition={{ duration: 2 }}
							aria-label="content"
						>
							<h1 className="text-5xl text-center">GET MY CV FOR FREE!</h1>
							<img
								src="/cv/download.svg"
								alt="download"
								className="w-10 h-10 mt-5"
							/>
						</m.div>
					</a>
				</section>
			</LazyMotion>
		</>
	);
}
