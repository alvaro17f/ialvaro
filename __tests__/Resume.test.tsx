import CV from "@/app/pages/CV";
import { cleanup, render, screen } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";

describe("<CV />", () => {
	beforeEach(() => {
		const mockIntersectionObserver = vi.fn();
		mockIntersectionObserver.mockReturnValue({
			observe: () => null,
			unobserve: () => null,
			disconnect: () => null,
		});
		window.IntersectionObserver = mockIntersectionObserver;
		render(<CV />);
	});
	afterEach(cleanup);

	it("should match the snapshot", () => {
		const { container } = render(<CV />);
		expect(container).toMatchSnapshot();
	});

	it("should render a heading", () => {
		expect(
			screen.getByRole("heading", { level: 1, name: /cv/i }),
		).toBeDefined();
	});
});
