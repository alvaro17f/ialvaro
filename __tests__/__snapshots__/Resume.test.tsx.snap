// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CV /> > should match the snapshot 1`] = `
<div>
  <section
    id="cv"
  >
    <div
      class="h-24"
    />
    <a
      href="/cv/cv.pdf"
      rel="noreferrer"
      target="_blank"
    >
      <div
        aria-label="content"
        class="grid place-items-center p-5 mb-5 rounded-xl border-[5px] border-alvaro-danger text-alvaro-white"
        style="opacity: 0;"
        tabindex="0"
      >
        <h1
          class="text-5xl text-center"
        >
          GET MY CV FOR FREE!
        </h1>
        <img
          alt="download"
          class="w-10 h-10 mt-5"
          src="/cv/download.svg"
        />
      </div>
    </a>
  </section>
</div>
`;
